Audio input

herkulano opened the pull request:
Moved to this branch for easier QA

cc/ @herkulano @kerzmaximilian @nithinth7

herkulano commented:
@nithinth7 there are some dependencies missing, including the open smile plugin. can you make changes so that we can build it?

```bash
[14:30:26]  typescript: src/components/audio-input/audio-input.ts, line: 34 
            Cannot find name 'MediaPlugin'. 

      L33:  compression: number
      L34:  media: MediaPlugin = null
      L35:  platform: boolean = false
```

//cc @kerzmaximilian

kerzmaximilian commented:
@herkulano can we go ahead and merge it?

herkulano commented:
@kerzmaximilian it's broken at the moment, can't even build it as is and we should test it before we merge it.

herkulano commented:
@nithinth7 I'me still getting the same error:
```
[14:40:51]  typescript: src/components/audio-input/audio-input.ts, line: 34 
            Cannot find name 'MediaPlugin'. 

      L33:  compression: number
      L34:  media: MediaPlugin = null
      L35:  platform: boolean = false
```

Can you try to do a fresh pull of the repo and install to reproduce this bug on your end and fix it?

//cc @kerzmaximilian

kerzmaximilian commented:
Thanks for checking @herkulano.

@nithinth7 could you have a look at this, please.

nithinth7 commented:
Hello @herkulano,
Could you please give me steps you are following? Because I am not able to reproduce the issue in my system. I will try again to reproduce the issue with your steps and fix it.

herkulano commented:
Hi @nithinth7 

The problem was on my side. I cleaned up the repo and I no longer have the issue in Typescript.

However, I now have a Java issue when doing `$ ionic run android`

Could you test on your side since I've also updated the dependencies?

```
BUILD FAILED

Total time: 26.488 secs

Error: /Users/Herc/Documents/Repos/RADAR-Questionnaire/platforms/android/gradlew: Command failed with exit code 1 Error output:
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
/Users/Herc/Documents/Repos/RADAR-Questionnaire/platforms/android/src/com/audeering/opensmile/androidtemplate/Config.java:21: error: diamond operator is not supported in -source 1.6
    List<OpenSmilePlugins> plugins = new ArrayList<>();
                                                   ^
  (use -source 7 or higher to enable diamond operator)
/Users/Herc/Documents/Repos/RADAR-Questionnaire/platforms/android/src/myjni/OpenSmile.java:88: error: diamond operator is not supported in -source 1.6
        ArrayList<OpenSmilePlugins> ans = new ArrayList<>();
                                                        ^
  (use -source 7 or higher to enable diamond operator)
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
```

herkulano commented:
Hi @nithinth7 

Sorry, I spoke too soon, after updating the dependencies and cleaning up the project again, I have now the same error as before:
```
[14:40:51]  typescript: src/components/audio-input/audio-input.ts, line: 34 
            Cannot find name 'MediaPlugin'. 

      L33:  compression: number
      L34:  media: MediaPlugin = null
      L35:  platform: boolean = false
```

herkulano commented:
@nithinth7 the steps to reproduce are:

Pull the changes I've committed, then
```
$ rm -rf node_modules plugins platforms 
$ npm i OR $ yarn
$ ionic state reset
$ ionic serve OR $ ionic run android
```

nithinth7 commented:
@herkulano 
The java error is causing by the old java version. Updating the java version in RADAR-Questionnaire\platforms\android\build.gradle to 1.7 will resolve the issue.
 From sourceCompatibility JavaVersion.VERSION_1_6 to sourceCompatibility JavaVersion.VERSION_1_7

Could you please execute the following commands in the same order? Because this is working for me.
        yarn
	ionic serve
	ionic state restore
	ionic serve
	       Change JavaVersion to 1.7
	ionic run android

herkulano commented:
@nithinth7 

From a fresh pull of the branch and install, I can't get past the `ionic serve` command with the error reported before.

@kerzmaximilian could you test it on your side with a fresh pull and install?

nithinth7 commented:
@herkulano Could you please execute the all commands even though you are getting error in the first 'ionic serve' command? 'Ionic state restore' command work properly only after at least one execution of  'ionic serve' command. For audio recording, we are using opensmile plugin, and the plugin should be installed before the execution of the application. I am using 'ionic state restore' command to install all the platforms and plugin required for the application. But 'ionic state restore' command not working when it is run before 'ionic serve'. So just run 'ionic serve' once, then execute 'ionic state restore'. Once the platform and plugins added, both 'ionic serve' and 'ionic run' should work as expected.

herkulano commented:
@nithinth7 I did, but it doesn't work. I would be surprised if running `ionic serve` had an influence on the `restore` and vice-versa but tried it anyway and it doesn't work. @kerzmaximilian will set up a call for us to go through this in detail.

nithinth7 commented:
@herkulano, 
It requires 3 plugins for audio recording questions - cordova-plugin-media, cordova-plugin-android-permissions, and cordova-plugin-opensmile. Adding these 3 plugins in package.json will fix the issue (It looks like cordova-plugin-media and cordova-plugin-android-permissions plugins are removed from package.json while merging master branch to audio-input branch)
